### ISSUE TITLE
Zero Loss Value When Training with SameDiff

### ISSUE THREAD

**Issue by AllenWGX** [deeplearning4j/deeplearning4j#9806]
#### Issue Description
I find that loss value always equals zero when using SameDiff to train. I print the trainable variable's gradient value and more log infomation. It shows that gradient's value is not zero, and I can also get the non-zero loss if I execute forward pass. Pls help to check the code and log infos.

        int nIn = 4;
        int nOut = 1;
        int NUM_SAMPLES = 300;
        int epoches = 2;
        int minibatch = 3;

        SameDiff sd = SameDiff.create();

        //First: Let's create our placeholders. Shape: [minibatch, in/out]
        SDVariable input = sd.placeHolder("input", DataType.FLOAT, -1, nIn);
        SDVariable labels = sd.placeHolder("labels", DataType.FLOAT, -1, nOut);

        //Second: let's create our variables
        SDVariable weights = sd.var("weights", new XavierInitScheme('c', nIn, nOut), DataType.FLOAT, nIn, nOut);
        SDVariable bias = sd.var("bias");

        //And define our forward pass:
        SDVariable out = input.mmul(weights).add(bias);     //Note: it's broadcast add here

        //And our loss function
        SDVariable mse = sd.loss.meanSquaredError("mse", labels, out, null);

        //Let's create some mock data for this example:
        Nd4j.getRandom().setSeed(12345);
        INDArray inputArr = Nd4j.rand(minibatch, nIn);
        INDArray labelArr = Nd4j.rand(minibatch, nOut);

        Map<String,INDArray> placeholderData = new HashMap<>();
        placeholderData.put("input", inputArr);
        placeholderData.put("labels", labelArr);

        //Execute forward pass:
        INDArray loss = sd.output(placeholderData, "mse").get("mse");
        System.out.println("MSE: " + loss);

        //Calculate gradients:
        Map<String,INDArray> gradMap = sd.calculateGradients(placeholderData, "weights");
        System.out.println("Weights gradient:");
        System.out.println(gradMap.get("weights"));
        
        //Mock random dataset for training
        INDArray indFeature = Nd4j.rand(new long[] {NUM_SAMPLES, nIn});
        INDArray indLabel = Nd4j.rand(new long[] {NUM_SAMPLES, nOut});
        DataSet ds = new DataSet(indFeature, indLabel);
        SplitTestAndTrain train_test = ds.splitTestAndTrain(0.7);
        DataSet dsTrain = train_test.getTrain();
        DataSet dsTest = train_test.getTest();
        DataSetIterator trainIter = new ListDataSetIterator<DataSet>(Lists.newArrayList(dsTrain), minibatch);
        DataSetIterator testIter = new ListDataSetIterator<DataSet>(Lists.newArrayList(dsTest), minibatch);
        //Train model
        double learningRate = 1e-3;
        TrainingConfig config = new TrainingConfig.Builder()                             
            .updater(new Sgd(learningRate))        
            .dataSetFeatureMapping("input")         
            .dataSetLabelMapping("labels")
            .build();
        sd.setTrainingConfig(config);
        sd.setListeners(new ScoreListener(1));
        sd.setLossVariables(mse);
        History hist = sd.fit(trainIter, epoches);
        System.out.println(hist.getLossCurve().getLossValues());

#### Version Information

* Deeplearning4j version: 1.0.0-SNAPSHOT & 1.0.0-beta7  & 1.0.0-M2.1
* Platform information (OS, etc): Win10
* CUDA version, if used: **Not Used**
* NVIDIA driver version, if in use: **Not Used**

#### Additional Information

Where applicable, please also provide:

* Full log or exception stack trace:https://gist.github.com/AllenWGX/8ab7e50530befeafefadd6cb9555d88a
* pom.xml file or similar:https://gist.github.com/AllenWGX/0577bd01a0cded0e1faff3c57969b374
